src/api/firebase/storage/storage-actions.ts

```diff
@@ -0,0 +1,44 @@
+import { v4 as uuidv4 } from 'uuid';
+
+import { storage, storageEventState, storageTaskState } from '../firebase.api';
+import { StoragePath, UploadTask } from '../firebase.types';
+
+const monitorBatchUpload = <T>(task: UploadTask, cb: (url: string) => Promise<T>) => {
```

Comment:
I would replace `cb` with `callback`, because you know what's going on right away.
`cb` actually means nothing.

Reply:
I am used to this notation, but I understand that for you the notation "callaback" is more readable. I've changed it.